client/src/App.tsx

```diff
@@ -16,6 +18,8 @@ import 'swiper/css';
 
 
 const App = () => {
+
+  useGlobalColor();
```

Comment:
Why you chose to use this `useGlobalColor()` inside the App component and not outside of it (at line 19 for example)

```suggestion
useGlobalColor();
const App = () => {

```

Its more of a thought experiment than an actual problem.

Reply:
to use it anytime the page is refreshed and the app is rerenderd, it is not the right approach?